src/components/coursCard.tsx

```diff
@@ -0,0 +1,57 @@
+import { coursDetails } from "@/types/coursDetails.type";
+import { FC } from "react";
+import { FaStar } from "react-icons/fa";
+import FlagAR from "@/assets/flag_ar.svg";
+import FlagEN from "@/assets/flag_en.svg";
+import Image from "next/image";
+import { FaRegStar } from "react-icons/fa";
+import { v4 as uuidv4 } from 'uuid';
+
+
+export const CoursCard:FC<{data: coursDetails}> = ({data})=>{
+    return <div className="w-[400px] h-[300px] border-black rounded-md border-[3px] flex flex-col">
+        {/* card title */}
+        <div className="bg-[#FFDF36] p-3 flex justify-between flex-row-reverse items-center border-black border-[3px] rounded-[4px] border-t-0 border-r-0 border-l-0 basis-1/6">
+            {data.title}
+            { data.level === "سهل" && <p className="rounded-xl border border-black py-1 px-2 bg-[#A2FF93]">{data.level}</p>}
+            { data.level === "متوسط" && <p className="rounded-xl border border-black py-1 px-2 bg-[#FFEE93]">{data.level}</p>}
+            { data.level === "صعب" && <p className="rounded-xl border border-black py-1 px-2 bg-[#FF9393]">{data.level}</p>}
+        </div>
+        {/* card body */}
+        <div className="text-right flex flex-col justify-between basis-2/3 p-3">
+            <p>{data.description}</p>
+            <div>
+                <p className="font-bold text-[30px]">{data.rating}</p>
+                <div className="flex justify-between items-center flex-row-reverse">
+                    <div className="flex flex-row-reverse items-center gap-2">
+                        <div className="flex flex-row-reverse">
+                            {Array.from(Array(5)).map((_, i)=>{
+                                if(i+1 <= data.rating)
+                                    return <FaStar color="#FFDF36" className="w-[30px] h-[30px]"/>;
+                                else
+                                    return <FaRegStar color="#FFDF36" className="w-[30px] h-[30px]"/>;
+                            })}
+                        </div>
+                        <p>({data.nbrOfRaters} تقيم)</p>
+                    </div>
+                    <div className="flex justify-center items-center gap-3">
+                        {data.languages.map((lang)=>{
+                            if(lang === "arabic")
+                                return <Image src={FlagAR} alt="flag" className="w-[30px] h-[30px]"/>
+                            else if(lang === "english")
+                                return <Image src={FlagEN} alt="flag" className="w-[30px] h-[30px]"/>
+                        })}
```

Comment:
thanks for your changes, i think you can apply this comment https://github.com/Warriors-of-Light/sahwa/pull/1#discussion_r1644620137 here, if we will manage different arabic dialects in the future, we will have a big block of if .. else .. . i prefer using an object with key language and value src
```suggestion
                        {data.languages.map((lang)=>{
                                return <Image src={yourObject[lang]} alt="flag" className="w-[30px] h-[30px]"/>
                        })}
```
what do you think ?

Reply:
ok it looks like a perfect solution to minimize the code of this block

Reply:
it's also quicker.

just for your knowledge. as in the worst case, we usually take the last condition so in term of complexity it's O(number of conditions) but accessing directly by key in an object is considered O(1) (o(n) in worst case if the object is full)